src/scales/scale.time.js

```diff
@@ -176,6 +176,19 @@ module.exports = function(Chart) {
 			Chart.Scale.prototype.initialize.call(this);
 		},
 
+        /**
+         * Allow x-axis to be accessed via x or t attribute
+         */
+		getRightValue: function(rawValue) {
+			if (typeof(rawValue) === 'object') {
+				if ((rawValue instanceof Date) || rawValue.isValid) {
```

Comment:
I agree that `core.scale.js` should not depend of moment, but I realize that we should remove these checks from `getRightValue` in `core.scale.js` and keep only: `object ? horizontal ? value.x : value.y`. There is no reason to have them in the base class, it's time scale specific and this new override is the way to go.

Since there is already a dependency of moment in `scale.time.js`, I prefer an explicit check which is more robust and readable (I'm also changing it in `parseTime`).